src/http-client.ts

```diff
@@ -40,14 +47,75 @@ export class HttpClient {
       .set('Authorization', `Bearer ${opts.authToken}`)
       .type(opts.contentType || 'json');
 
-    return response.send(opts.data); // trigger request
+    return response.send(opts.data) // trigger request
+      .catch(err => { throw this.formatRequestError(err); });
   }
 
   stopAllRetries (): void {
     Array.from(this._httpRetryingRequests.keys())
       .forEach(key => this._cancelAndRemoveValueFromRetryMap(key));
   }
 
+  formatRequestError (error: Error | ISuperagentNetworkError | ISuperagentResponseError): IError | INetworkError | IResponseError {
+    /* if network error */
+    if (this.isSuperagentNetworkError(error)) {
+      return {
+        status: error.status,
+        method: error.method,
+        url: error.url,
+        crossDomain: error.crossDomain,
+        message: error.message,
+        name: error.name,
+        stack: error.stack
+      };
+    }
+
+    /* if superagent response error */
+    if (this.isSuperagentResponseError(error)) {
+      const res = error.response;
+      return {
+        status: error.status,
+
+        correlationId: res.headers['inin-correlation-id'],
+        responseBody: res.text,
+
+        method: res.req.method,
+        requestBody: res.req._data,
+
+        url: res.error.url,
+        message: res.error.message,
+        name: res.error.name,
+        stack: res.error.stack
+      };
+    }
+
+    /* if we don't have a superagent error */
+    return {
```

Comment:
is there a reason to not just return `Error` here?

Reply:
I was torn here. My thought was: this function wants to format all errors, not just `superagent` errors. By formatting the error this way, it will hit the console a lot cleaner.

Reply:
Albeit, this does mess up the prototype chain making `error instanceof Error` return `false`. I can switch it back if you think that would be helpful

Reply:
I think we ought to just leave it as an error, but I don't have particularly strong feelings either way.